conn.go

```diff
@@ -120,9 +120,11 @@ type Conn struct {
 
 	headerBuf []byte
 
-	uniq  chan int
-	mu    sync.RWMutex
-	calls map[int]*callReq
+	availableStreams int32
```

Comment:
I think it would be best to move the stream ID related things into its own struct as conn is starting to get quite large.